### PR TITLE
(refactor) some opportunities to use lambda functions

### PR DESCRIPTION
**Pull request by desmorto**

### CODE REVIEW

File: src/java/org/apache/cassandra/auth/CassandraRoleManager.java [hunk 1]
```diff
@@ -448,10 +449,6 @@ private void removeAllMembers(String role) throws RequestValidationException, Re
                 consistencyForRole(role));
     }
 
-    /*
-     * Convert a map of Options from a CREATE/ALTER statement into
-     * assignment clauses used to construct a CQL UPDATE statement
-     */
```
Comment (clohfink) [inline src/java/org/apache/cassandra/auth/CassandraRoleManager.java:454]: why delete this comment?

File: src/java/org/apache/cassandra/cql3/CQL3Type.java [hunk 2]
```diff
@@ -826,13 +827,18 @@ public CQL3Type prepare(String keyspace, Types udts) throws InvalidRequestExcept
...
+                                t -> {
+                                    if (t.isCounter())
+                                        throw new InvalidRequestException(
+                                                "Counters are not allowed inside tuples");
+                                    return t;
+                                })
+                        .forEach(
+                                t -> {
+                                    ts.add(t.prepare(keyspace, udts).getType());
+                                });
```
Comment (clohfink) [inline src/java/org/apache/cassandra/cql3/CQL3Type.java:841]: This looks worse, and if your going to convert to a new list you can just do the mapping like

```
types.stream().map(t ->
{
    if (t.isCounter())
        throw new InvalidRequestException("Counters are not allowed inside tuples");
    return t.prepare(keyspace, udts).getType()
}
```

**Review by clohfink — CHANGES_REQUESTED** [review]
These seem auto generated without any reasoning. Theres no performance improvement and after converted to lambda some actually make less sense (in cases where building up list inside lambda instead of doing it functionally with map/reduce)

Also if going to use tooling should refactor to call single function lambdas by function name to make it more clear (ie Raw::supportsFreezing instead of t -> t. supportsFreezing()).

-  The style guideline says that `{`'s should start on following line, not same line. http://cassandra.apache.org/doc/latest/development/code_style.html#general-code-conventions
- The style guideline dictates import order which this changes (for no reason) to be different. http://cassandra.apache.org/doc/latest/development/code_style.html#imports

There should be a JIRA or email to discuss if this is even a good idea

**Comment by jeffjirsa** [comment]
Please see my comment to you in #155  - this type of refactor is rarely desirable, and needs to go through JIRA. We have docs on how to contribute, and opening this type of pull request is not generally useful.

**Comment by smiklosovic** [comment]
Given this is 5 years old and without ticket I am closing this PR. Feel free to raise this in mailing list or file a ticket etc.

**Pull request closed without merging**